libs/agno/agno/workflow/v2/condition.py

```diff
@@ -25,41 +34,413 @@
 class Condition:
     """A condition that executes a step (or list of steps) if the condition is met"""
 
-    # Evaluator can be a async or sync function, or a boolean
-    # If it is a function, it has to return the step/steps to execute
-    # If it is a boolean, it will be used to determine if all the provided steps should be executed
-    evaluator: Union[Callable[[Any], Union[bool, Step, List[Step], Awaitable[Step], Awaitable[List[Step]]]]]
+    # Evaluator should only return boolean
+    evaluator: Union[
+        Callable[[StepInput], bool],
+        Callable[[StepInput], Awaitable[bool]],
+        bool,
+    ]
     steps: WorkflowSteps
 
     name: Optional[str] = None
     description: Optional[str] = None
 
+    def _prepare_steps(self):
+        """Prepare the steps for execution - mirrors workflow logic"""
+        from agno.agent.agent import Agent
+        from agno.team.team import Team
+        from agno.workflow.v2.parallel import Parallel
+        from agno.workflow.v2.step import Step
+
+        prepared_steps = []
+        for step in self.steps:
+            if isinstance(step, Callable):
+                prepared_steps.append(Step(name=step.__name__, description="User-defined callable step", executor=step))
+            elif isinstance(step, Agent):
+                prepared_steps.append(Step(name=step.name, description=step.description, agent=step))
+            elif isinstance(step, Team):
+                prepared_steps.append(Step(name=step.name, description=step.description, team=step))
+            elif isinstance(step, (Step, Loop, Parallel, Condition)):
```

Comment:
```suggestion
            elif isinstance(step, (Step, Loop, Parallel, Condition, Router)):
```

We should probably store this list of types somewhere as a type alias to make sure we don't miss a reference somewhere